publish-token/src/main/java/fi/thl/covid19/publishtoken/generation/v1/PublishTokenGenerationController.java

```diff
@@ -48,7 +48,12 @@ public PublishToken generateToken(@RequestHeader(name = SERVICE_NAME_HEADER) Str
                     requestService,
                     request.requestUser,
                     request.symptomsExist);
-            request.patientSmsNumber.ifPresent(number -> smsService.send(number, token));
+            request.patientSmsNumber.ifPresent(number -> {
+                        if (!smsService.send(number, token)) {
+                            throw new IllegalStateException("SMS send failed");
```

Comment:
Certainly a clean implementation. However, this does mean that the user has no other option than to try again. If there's an issue in the SMS service, there's no way to get out a valid code. Also, since the code is already generated at this point (just not returned), the user might aggressively retry, leading to many valid codes in backend. Of course, they'll expire pretty soon, so it's likely not a big deal.

Overall, I guess this is fine if SMS-service errors are expected to be rare and short-lived. It would be nicer to return both the code and SMS status, but that's obviously a much bigger change and requires changes on the caller side as well.

Reply:
Yep, however user can create code without sms-number even if there is issues with sms-service. Anyway, while with human users, it's probably not a big problem have some unnecessary tokens, with bot-clients it could be. So, I have added invalidation functionality now for erroneus tokens.

So, in case there is error when sending sms user needs to:

a) try again and wait the problem to be fixer
b) try again without phone number and deliver token to end-user with other means

I have also added error counter for more robust monitoring in server side.